lib/rucio/core/replica.py

```diff
@@ -1637,6 +1638,98 @@ def list_unlocked_replicas(rse, limit, bytes=None, rse_id=None, worker_number=No
     return rows
 
 
+@transactional_session
+def list_and_mark_unlocked_replicas(limit, bytes=None, rse_id=None, worker_number=None, total_workers=None, delay_seconds=0, session=None):
+    """
+    List RSE File replicas with no locks.
+
+    :param limit:              Number of replicas returned.
+    :param bytes:              The amount of needed bytes.
+    :param rse_id:             The rse_id.
+    :param total_workers:      Number of total workers.
+    :param worker_number:      id of the executing worker.
+    :delay_seconds:            The delay to query replicas in BEING_DELETED state
+    :param session:            The database session in use.
+
+    :returns: a list of dictionary replica.
+    """
+    replicas_alias = aliased(models.RSEFileAssociation, name='replicas_alias')
+
+    # filter(models.RSEFileAssociation.state != ReplicaState.BEING_DELETED).\
+    none_value = None  # Hack to get pep8 happy...
+    query = session.query(replicas_alias.scope, replicas_alias.name, replicas_alias.path, replicas_alias.bytes, replicas_alias.tombstone, replicas_alias.state, func.count(models.RSEFileAssociation.scope)).\
```

Comment:
This query should be with `with_for_update(skip_locked=True)` 
We need to do retrys though if this throws an exception